pkgs/shells/zsh/oh-my-zsh/default.nix

```diff
@@ -35,6 +35,13 @@ stdenv.mkDerivation rec {
          -e 's/\# \(DISABLE_AUTO_UPDATE="true"\)/\1/' \
    $template
 
+  chmod -R +w lib
+
+  # Remove functions that don't work on Nix
+  sed -i -e "/uninstall_oh_my_zsh()/,/}/ d" \
```

Comment:
This will probably break as soon as this function has a `}` (in a future version) leaving a broken file in `$out`. I'd really prefer to apply a patch here.

Reply:
It might be better to add
```
  unfunction uninstall_oh_my_zsh
  unfunction upgrade_oh_my_zsh
```
to $template, that way we wouldn't need to update a patch file after changes to the function are made and it is a simpler addition than adding a patch file.

Reply:
This is probably the best way to go about it, since patching the sources is inherently fragile.  You should be able to add those two lines just after `cat >> $template <<- EOF`.

I didn't know `unfunction` even existed, that's a cool name :slightly_smiling_face: